tests/tekton-resources/tasks/generators/clusterloader/load-pod-identity.yaml

```diff
@@ -175,6 +194,81 @@ spec:
       S3_RESULT_PATH=$(cat $(results.s3_result.path))
       echo "S3 Path: $S3_RESULT_PATH"
       aws sts get-caller-identity
+
+      REGION=$(params.region)
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+
+      CLUSTER_NAME=$(params.cluster-name)
+      NAMESPACE=$(params.cl2-metric-namespace)
+      DIMENSION_NAME=$(params.cl2-metric-dimension-name)
+      DIMENSION_VALUE=$CLUSTER_NAME
+      METRIC_LATENCY_NAME=$(params.cl2-metric-latency-name)
+      PERIOD=$(params.cl2-metric-period)
+
+      # since the scalability test is running with the same cluster name, with cluster recreation
+      # it is important to know the range of start and end time to query metrics for the current run
+      # here we use cluster creation time start as start time and the current time as end time
+      START_TIME=$(aws eks $ENDPOINT_FLAG --region $REGION describe-cluster \
+        --name "$CLUSTER_NAME" \
+        --query "cluster.createdAt" \
+        --output text)
+
+      END_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --extended-statistics p50 p90 p99 \
+        --output json)
+
+      # extract p50 p90 p99 of credential fetching
+      latest=$(echo "$response" | jq -r '.Datapoints | sort_by(.Timestamp) | last')
+      p50=$(echo "$latest" | jq -r '."ExtendedStatistics"."p50" // "N/A"')
+      p90=$(echo "$latest" | jq -r '."ExtendedStatistics"."p90" // "N/A"')
+      p99=$(echo "$latest" | jq -r '."ExtendedStatistics"."p99" // "N/A"')
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --statistics SampleCount \
+        --output json)
+
+      total_samples=$(echo "$response" | jq '[.Datapoints[].SampleCount] | add // 0')
```

Comment:
We need to capture what  is the `rate` at which credentials are fetched from service. To be able to compare that against the scheduler throughput.

Currently you have some kind of measuring by setting the timeout  https://github.com/awslabs/kubernetes-iteration-toolkit/pull/512/files#diff-d2d660edac904aa96e330bfae7bf67ef6885190877c5ab7668f0f157057da03fR61 accordingly by computing total number of pods and client pod creation rate/scheduler rate .

This will only give some kind of approximation  but not fully accurate throughput of your service using this kind of measurement.